### ISSUE TITLE
Got Error(symbol(s) not found for architecture x86_64)  brew install solidity

### ISSUE THREAD

**Issue by wudongyang** [ethereum/homebrew-ethereum#164]
```
==> Installing solidity from ethereum/ethereum
==> Downloading https://github.com/ethereum/solidity/releases/download/v0.4.22/solidity_0.4.22.tar.gz
Already downloaded: /Users/wdy/Library/Caches/Homebrew/solidity-0.4.22.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/solid
==> make install
Last 15 lines from /Users/wdy/Library/Logs/Homebrew/solidity/02.make:
Undefined symbols for architecture x86_64:
  "Json::Value::append(Json::Value&&)", referenced from:
      dev::solidity::CompilerStack::createMetadata(dev::solidity::CompilerStack::Contract const&) const in libsolidity.a(CompilerStack.cpp.o)
      dev::solidity::ASTJsonConverter::appendMove(Json::Value&, Json::Value&&) in libsolidity.a(ASTJsonConverter.cpp.o)
      dev::solidity::ASTJsonConverter::visit(dev::solidity::SourceUnit const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::toJson<dev::solidity::ASTNode>(std::__1::vector<std::__1::shared_ptr<dev::solidity::ASTNode>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::ASTNode> > > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      dev::solidity::ASTJsonConverter::visit(dev::solidity::PragmaDirective const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::getContainerIds<std::__1::vector<dev::solidity::ContractDefinition const*, std::__1::allocator<dev::solidity::ContractDefinition const*> > >(std::__1::vector<dev::solidity::ContractDefinition const*, std::__1::allocator<dev::solidity::ContractDefinition const*> > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      Json::Value dev::solidity::ASTJsonConverter::toJson<dev::solidity::InheritanceSpecifier>(std::__1::vector<std::__1::shared_ptr<dev::solidity::InheritanceSpecifier>, std::__1::allocator<std::__1::shared_ptr<dev::solidity::InheritanceSpecifier> > > const&) in libsolidity.a(ASTJsonConverter.cpp.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [solc/solc] Error 1
make[1]: *** [solc/CMakeFiles/solc.dir/all] Error 2
make: *** [all] Error 2

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/ethereum/homebrew-ethereum/issues
```

**Comment by wudongyang** [comment]
OS: MAC 


$ geth version
Geth
Version: 1.8.3-stable
Architecture: amd64
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.10.1
Operating System: darwin
GOPATH=
GOROOT=/usr/local/opt/go/libexec

**Comment by axic** [comment]
@chfast seems like to CMake fixes didn't fix it yet. Same problem as #163.

**Comment by axic** [comment]
Well it is also another problem: it still seems the host installed jsoncpp takes precendence over the downloaded one. Note, the move semantics are only used from jsoncpp 1.8.4 onwards, and the current downloaded version in the release is 1.7.7. As such the above log means the user has 1.8.4 installed and that takes precedence.

**Comment by wudongyang** [comment]
Thx

fix it with "brew uninstall --ignore-dependencies jsoncpp"

**Comment by jasoncook26** [comment]
Thx

**Comment by axic** [comment]
Will be completely fixed in the next release after 0.4.23.
